server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionCoordinator.java

```diff
@@ -0,0 +1,766 @@
+              try {
+                client = getTabletServerConnection(tsi);
+                LOG.debug(
+                    "Checking to see if tserver {} is running external compaction for extent: {}",
+                    tsi.getHostAndPort(), extent);
+                boolean tserverMatch = client.isRunningExternalCompaction(TraceUtil.traceInfo(),
+                    getContext().rpcCreds(), job.getExternalCompactionId(), job.getExtent());
+                if (tserverMatch) {
+                  LOG.debug(
+                      "Tablet server {} is running external compaction for extent: {}, adding to running list",
+                      tsi.getHostAndPort(), extent);
+                  RUNNING.put(ExternalCompactionId.of(job.getExternalCompactionId()),
+                      new RunningCompaction(job, ExternalCompactionUtil.getHostPortString(hp),
+                          tsi));
+                  matchFound = true;
+                }
+              } catch (TException e) {
+                LOG.error(
+                    "Error from tserver {} while trying to check if external compaction is running, trying next tserver",
+                    ExternalCompactionUtil.getHostPortString(tsi.getHostAndPort()), e);
+              } finally {
+                ThriftUtil.returnClient(client);
+              }
+            }
+          }
+
+          if (!matchFound) {
+            LOG.warn(
+                "There is an external compaction running on a compactor, but could not find corresponding tablet server. Extent: {}, Compactor: {}, Compaction: {}",
+                extent, hp, job);
+          }
+        });
+      }
+      tservers.clear();
+    } else {
+      LOG.info("No running tablet servers found, continuing startup");
+    }
+
+    tserverSet.startListeningForTabletServerChanges();
+    startDeadCompactionDetector();
+
+    LOG.info("Starting loop to check tservers for compaction summaries");
+    while (!shutdown) {
+      long start = System.currentTimeMillis();
+      tserverSet.getCurrentServers().forEach(tsi -> {
+        try {
+          TabletClientService.Client client = null;
+          try {
+            LOG.debug("Contacting tablet server {} to get external compaction summaries",
+                tsi.getHostPort());
+            client = getTabletServerConnection(tsi);
+            List<TCompactionQueueSummary> summaries =
+                client.getCompactionQueueInfo(TraceUtil.traceInfo(), getContext().rpcCreds());
+            summaries.forEach(summary -> {
+              QueueAndPriority qp =
+                  QueueAndPriority.get(summary.getQueue().intern(), summary.getPriority());
+              synchronized (qp) {
+                TIME_COMPACTOR_LAST_CHECKED.computeIfAbsent(qp.getQueue(), k -> 0L);
+                QUEUE_SUMMARIES.update(tsi, summaries);
+              }
+            });
+          } finally {
+            ThriftUtil.returnClient(client);
+          }
```

Comment:
Wondering if these RPCs should be done in a thread pool so that multiple tserver can be contacted in parallel.

Reply:
Done in b59c13ff80da6491a89757a00c0fa32166d038d2